services/src/main/java/com/linkedin/kmf/services/TopicManagementService.java

```diff
@@ -182,7 +184,7 @@ public void run() {
           if (topicState.replicationFactor() == -1) {
             return;
           }
-          if (topicState.someBrokerMissingPartition()) {
+          if (topicState.someBrokerMissingPartition() && !topicAssignmentIsRunning()) {
```

Comment:
I agree it makes sense to check if there is ongoing reassignment. But do we still need to call `waitForTopicAssignmentsToComplete()` after this check?